templates/aws-tap-entrypoint-existing-vpc.template.yaml

```diff
@@ -1374,11 +1372,21 @@ Resources:
                   - eks:DescribeCluster
                   - eks:DescribeNodegroup
                 Resource:
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-run
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-build
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-iterate
-                  - !Sub arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterName}-view
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterNameSingle}
+                      - EKSClusterNameSingle: !Join ['-', [!FindInMap [ EKSClusterNames, Single, Name ], !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterNameRun}
+                      - EKSClusterNameRun: !Join ['-', [!FindInMap [ EKSClusterNames, Run, Name ], !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterNameBuild}
+                      - EKSClusterNameBuild: !Join ['-', [!FindInMap [ EKSClusterNames, Build, Name ], !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterNameIterate}
+                      - EKSClusterNameIterate: !Join ['-', [!FindInMap [ EKSClusterNames, Iterate, Name ], !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
+                  - !Sub
+                      - arn:${AWS::Partition}:eks:${AWS::Region}:${AWS::AccountId}:cluster/${EKSClusterNameView}
+                      - EKSClusterNameView: !Join ['-', [!FindInMap [ EKSClusterNames, View, Name ], !Select [4, !Split ['-', !Select [2, !Split ['/', !Ref AWS::StackId]]]]]]
```

Comment:
FWIW: it would be nicer (we'd get rid of the `!Sub`), if we could get the ARN of a cluster directly from an EKS stack output, but checking real quick seems to suggest that this is not exposed.